src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/ReflectionUtils/MetadataLoadContext.cs

```diff
@@ -18,14 +18,14 @@ public MetadataLoadContext(Compilation compilation)
         {
             _compilation = compilation;
             Dictionary<AssemblyName, IAssemblySymbol> assemblies = compilation.References
-                                        .OfType<PortableExecutableReference>()
-                                        .ToDictionary(r => AssemblyName.GetAssemblyName(r.FilePath),
-                                                      r => (IAssemblySymbol)compilation.GetAssemblyOrModuleSymbol(r)!);
+                .OfType<PortableExecutableReference>()
+                .ToDictionary(
+                    r => string.IsNullOrWhiteSpace(r.FilePath) ?
+                        new AssemblyName(r.Display) : AssemblyName.GetAssemblyName(r.FilePath),
+                    r => (IAssemblySymbol)compilation.GetAssemblyOrModuleSymbol(r)!);
 
             foreach (var item in assemblies)
             {
-                // REVIEW: We need to figure out full framework
```

Comment:
How is this TODO resolved?

Reply:
Currently using names instead of full names satisfies our requirement.
I believe getting the full name might require construction using syntax nodes.
Will resolve for now.

Issue link: https://github.com/dotnet/runtimelab/issues/48